step-capstone/src/components/ItemForm.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react'
+import {
+    Card,
+    CardActions,
+    CardContent,
+    Button,
+    Paper,
+    Tabs,
+    Tab,
+} from '@material-ui/core';
+import AddEventHotel from './AddEventHotel'
+import AddFlight from './AddFlight'
+
+
+export default class ItemForm extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isNewItem: props.data === undefined,
+            value: 0,
+            data: props.data,
+            isValidated: false
+        }
+        this.handleToggleTab = this.handleToggleTab.bind(this);
+        this.handleDataChange = this.handleDataChange.bind(this);
+        this.handleSave = this.handleSave.bind(this);
+        this.handleToggleValidation = this.handleToggleValidation.bind(this);
+    }
+
+    // clears data when tab is changed
+    handleToggleTab(event, newVal) {
+        this.setState({
+            value: newVal,
+            data: undefined
+        });
+    }
+
+    handleDataChange(newData) {
+        this.setState({
+            data: newData
+        })
+    }
+
+    // sets validation state to indicate whether or not all required inputs are filled out
+    handleToggleValidation(validated) {
+        if (validated !== this.state.isValidated) {
+            this.setState({
+                isValidated: validated
+            })
+        }
+    }
+
+    getForm() {
+        let isNew = this.state.isNewItem;
+        if ((!isNew && this.state.data.type === "event") || (isNew && this.state.value === 0)) {
+            return (
+                <AddEventHotel
+                    onDataChange={this.handleDataChange}
+                    type="event"
+                    data={this.state.data}
+                    onToggleValidation={this.handleToggleValidation}
+                />
+            )
+        } else if ((!isNew && this.state.data.type === "flight") || (isNew && this.state.value === 1)) {
+            return (
+                <AddFlight
+                    onDataChange={this.handleDataChange}
+                    data={this.state.data}
```

Comment:
Is `type="flight"` missing?

Reply:
It seemed unnecessary to pass down the type since in creating an AddFlight component the type must be flight. I passed down the type into event hotel since it could either be an event or a hotel.